Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/AbilData.xml

```diff
@@ -9950,6 +10115,29 @@
         </InfoArray>
         <Alert value="MorphComplete_Zerg"/>
     </CAbilTrain>
+    <CAbilTrain id="AP_MorphlingToGuardianSCBW@Morph">
+        <EditorCategories value="Race:Zerg,AbilityorEffectType:Units"/>
+        <RefundFraction>
+            <Resource index="Minerals" value="-0.75"/>
+            <Resource index="Vespene" value="-0.75"/>
+            <Resource index="Terrazine" value="-0.75"/>
+            <Resource index="Custom" value="-0.75"/>
+        </RefundFraction>
+        <Activity value="UI/Morphing"/>
+        <ActorKey value="GuardianAspect"/>
+        <Flags index="KillOnFinish" value="1"/>
+        <Flags index="Select" value="1"/>
+        <Flags index="WaitForFood" value="0"/>
+        <MorphUnit value="AP_GuardianSCBWEgg"/>

```

Comment:
I tested that and didn't find it too weird, but I agree it's worth discussing!

Tradeoff is this:
- If we use normal egg, there's no hatching actor, so the air units will just kinda pop into existence
- If we use air egg, they look a little strange on the ground, but they hatch properly

Third option: we add a height modifier when the air egg is morphed from a morphling, so it floats a little higher off the ground. Not high enough to be an air unit (actors can't do that, and adding another morph behavior would make this too complex imo) but high enough that it doesn't clip through the ground, with the visual suggestion that it's going to be airborne soon.

Reply:
Well, some SC2's transition anims are worse than SC1's ones

Reply:
Maybe? That's subjective, and I don't understand how it impacts the discussion...

Reply:
Ultralisks also hatch in normal eggs

Reply:
Yes, because if you could figure out what unit was coming out of larvae it would potentially be a disadvantage in multiplayer. That doesn't apply to cocoons in multiplayer because each unit has max 1 advanced morph; if you see a bunch of cocoons in a roach/hydra army, you know those cocoons are going to turn into ravagers and lurkers, and that's why they're allowed to have their own dedicated egg model. Because having special models is _cooler._ And because there's no good reason not to.